### DISCUSSION TITLE
Stock Item custom value

### DISCUSSION THREAD

**Discussion by rellisdev** [inventree/InvenTree#8051] [Ideas]
I am using inventree to manage a large amount of stock, I have a case where a part (camera) is being used for multiple different jobs so having to find a way to differentiate between different stock items. The only way I have found so far to differentiate between stock items is with the serial tag, which is very useful, but I need something more so that I can include more details about each unique stock item.

My current method is that a stock item has it's serial number and we have a separate sheet for what serial numbers are assigned to what jobs. What would be really useful is a custom field for a stock item where I can put the job it is apart of so we can then print labels to put on the item with the name, serial and the job it is assigned to.

**Comment by SchrodingersGat** [comment]
@rellisdev there are a couple of ways you could go about this:

### Notes Field

You can use the "notes" field for each item to add some arbitary notes about each stock item

### Test Results

You can overload the "test results" functionality to add a "result" against each stock item which contains the custom data.

### Metadata

Each stock item can have custom "metadata" which allows you to add custom JSON data against the item - https://docs.inventree.org/en/stable/extend/plugins/metadata/

This is not generally available in the user interface, but can be used by plugins.

**Reply by mihail-cornev** [reply]
> https://docs.inventree.org/en/stable/extend/plugins/metadata/

hello. where can i find this plugin?

**Reply by matmair** [reply]
That is not a plugin but a core mechansim https://docs.inventree.org/en/stable/plugins/metadata/
